qiskit_nature/circuit/library/initial_states/fermionic_gaussian_state.py

```diff
@@ -0,0 +1,165 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Fermionic Gaussian states."""
+
+from typing import Optional, Sequence
+
+import numpy as np
+from qiskit import QuantumCircuit, QuantumRegister
+from qiskit_nature.converters.second_quantization import QubitConverter
+from qiskit_nature.mappers.second_quantization import JordanWignerMapper
+
+from .utils.givens_rotations import _prepare_fermionic_gaussian_state_jordan_wigner
+
+
+def _validate_transformation_matrix(
+    mat: np.ndarray, rtol: float = 1e-5, atol: float = 1e-8
+) -> None:
+    if not len(mat.shape) == 2:
+        raise ValueError("transformation_matrix must be a 2-dimensional array.")
```

Comment:
I think all of these errors could be improved if you provide the current value. I.e. something like the below:
```suggestion
        raise ValueError(f"transformation_matrix must be a 2-dimensional array, not {len(mat.shape)}-dimensional.")
```

Reply:
Similarly for other errors further down

Reply:
Done.